www/js/main.js

```diff
@@ -72,6 +73,28 @@ define([
 	console.log(_model);
 
 	_session.getModel(query).then(function(solutionGraph){
+		var checkBrowser = _session.browser.name;
+		var checkVersion = _session.browser.version;
+		//TODO :Check versions for topomath compatibility
+		if((checkBrowser ==="Chrome" && checkVersion<41) ||
+			(checkBrowser==="Safari" && checkVersion<8) ||
+			(checkBrowser==="msie" && checkVersion<11) ||
+			(checkBrowser==="Firefox") || (checkBrowser==="Opera")){
+
+			var errorMessage = new messageBox("errorMessageBox", "warn","You are using "+ _session.browser.name+
+				" version "+_session.browser.version +
+				". Topomath is known to work well in these (or higher) browser versions: Google Chrome v41 or later Safari v8 or later Internet Explorer v11 or later");
+			// adding close callback to update the state for browser message
+			var compatibiltyState = new State(query.u, query.s, "action");
+			errorMessage.addCallback(function(){
+				compatibiltyState.put("browserCompatibility", "ack_" + getVersion());
+			});
+			compatibiltyState.get("browserCompatibility").then(function(res) {
+				if(!(res && res == "ack_" + getVersion())){
+					errorMessage.show();
+				}
+			});
```

Comment:
this code has session info, and should be moved to session.js. It was used in Dragoon please check that.